pyensembl/genome.py

```diff
@@ -0,0 +1,848 @@
+# Copyright (c) 2015. Mount Sinai School of Medicine
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Contains the Genome class, with its millions of accessors and wrappers
+around an arbitrary genomic database.
+"""
+
+from __future__ import print_function, division, absolute_import
+
+from glob import glob
+import logging
+from os.path import join
+from os import remove
+
+from .common import (
+    require_human_transcript_id,
+    require_human_protein_id,
+    memoize
+)
+from .compute_cache import cached_object
+from .database import Database
+from .exon import Exon
+from .gene import Gene
+from .gtf import GTF
+from .sequence_data import SequenceData
+from .transcript import Transcript
+
+class Genome(object):
+    """
+    Bundles together the genomic annotation and sequence data associated with
+    a particular genomic database source (e.g. a single Ensembl release) and
+    provides a wide variety of helper methods for accessing this data.
+    """
+    def __init__(self,
+                 reference_name,
+                 genome_source,
+                 name=None,
+                 version=None,
+                 only_human=False,
+                 auto_download=False,
+                 local_fasta_filename_func=None,
+                 require_ensembl_ids=True):
+        self.reference_name = reference_name
+        self.genome_source = genome_source
+        self.name = name
+        self.version = version
+        self.only_human = only_human
+        self.auto_download = auto_download
+
+        # GTF object wraps the source GTF file from which we get
+        # genome annotations. Presents access to each feature
+        # annotations as a pandas.DataFrame.
+        self.gtf = GTF(
+            genome_source,
```

Comment:
Discussed offline: `genome_source` is also used for the error string. For now, we'll pass in both the path/URL and the `genome_source`.